interfaces/devdoc/uuid_requirements.md

```diff
@@ -0,0 +1,61 @@
+uuid requirements
+=================
+
+## Overview
+The uuid module generates unique IDs.
+
+## References
+
+[RFC 4122](https://datatracker.ietf.org/doc/html/rfc4122)
+
+## Exposed API
+```C
+
+typedef unsigned char UUID_T[16]; /*introduces UUID_T as "array of 16 bytes"*/
+
+    MOCKABLE_FUNCTION(, int, uuid_produce, UUID_T, destination);

```

Comment:
we also have uuid to/from string in c-util, will that stay in c-util?

Reply:
No, that's the whole point = move the generation to c_pal (and only the generation). The rest (to/from string) can stay in c_util.